power_grid_model_c/power_grid_model_c/include/power_grid_model_c/basics.h

```diff
@@ -137,6 +137,12 @@ typedef struct PGM_WritableDataset PGM_WritableDataset;
  * @brief Opaque struct for the information of the dataset.
  */
 typedef struct PGM_DatasetInfo PGM_DatasetInfo;
+
+/**
+ * @brief Opaque struct for the multi dimensional dataset class.
+ *
+ */
+typedef struct PGM_MultiDimensionalDataset PGM_MultiDimensionalDataset;
```

Comment:
maybe following the conventions of `std::mdspan`:


```suggestion
typedef struct PGM_MDDataset PGM_MDDataset;
```